stan/math/prim/arr/functor/mpi_cluster.hpp

```diff
@@ -157,7 +156,7 @@ struct mpi_cluster {
       // lock on the workers the cluster as MPI commands must be
       // initiated from the root and any attempt to do this on the
       // workers must fail
-      std::unique_lock<std::mutex> worker_lock(in_use_);
+      std::unique_lock<std::mutex> worker_lock(in_use());
```

Comment:
Did you guys already talk about why we're using `unique_lock` instead of `lock_guard`?

Reply:
I didn't even realize there was a `lock_guard`!  So no.

It looks like we should be using `std::lock_guard` unless we need something more specific or customizable.  

Here's [a really nice summary from Jaka's corner](http://jakascorner.com/blog/2016/02/lock_guard-and-unique_lock.html).  Thanks, Jaka!